File: src/main/java/com/keygenqt/mylibrary/api/controllers/IndexController.java
package com.keygenqt.mylibrary.api.controllers;

import com.keygenqt.mylibrary.api.assemblers.IndexResourceAssembler;
import com.keygenqt.mylibrary.api.resources.IndexResource;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/")
class IndexController {

    private final IndexResourceAssembler assembler;

    @Autowired
    public IndexController(IndexResourceAssembler assembler) {
        this.assembler = assembler;
    }

    @RequestMapping(method = RequestMethod.GET)
    ResponseEntity<IndexResource> index() {
        return ResponseEntity.ok(assembler.buildIndex());
    }
}

File: src/main/kotlin/com/keygenqt/mylibrary/api/IndexController.kt
package com.keygenqt.mylibrary.api

import com.keygenqt.mylibrary.api.resources.*
import org.springframework.beans.factory.annotation.*
import org.springframework.http.*
import org.springframework.web.bind.annotation.*
import org.springframework.web.bind.annotation.RequestMethod.*

@RestController
@RequestMapping("/")
class IndexController @Autowired constructor(private val indexResourceAssembler: IndexResourceAssembler) {
    @RequestMapping(method = [GET]) fun index(): ResponseEntity<IndexResource> {
        return ResponseEntity.ok(indexResourceAssembler.buildIndex())
    }
}